guides/plugins/plugins/checkout/payment/add-payment-plugin.md

```diff
@@ -342,88 +458,180 @@ class ExamplePayment implements RefundPaymentHandlerInterface
+A full-fledged Subscriptions feature with recurring payments is available exclusively through our [paid plans](https://www.shopware.com/en/pricing/).
 :::
 
-```php
-// <plugin root>/src/ExamplePayment.php
+::: info
+Usually, a billing agreement between the customer and the payment provider is necessary to allow recurring payments.
+:::
+
+::: code-group
+
+```php [MyCustomPaymentHandler.php]
 <?php declare(strict_types=1);
 
 namespace Swag\BasicExample\Service;
 
-use Shopware\Core\Checkout\Payment\PaymentException;
-use Shopware\Core\Checkout\Payment\Exception\RecurringPaymentProcessException;
-use Shopware\Core\Checkout\Payment\Cart\PaymentHandler\RecurringPaymentHandlerInterface;
-use Shopware\Core\Checkout\Payment\Cart\RecurringPaymentTransactionStruct;
 use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStateHandler;
+use Shopware\Core\Checkout\Payment\Cart\PaymentHandler\AbstractPaymentHandler;
+use Shopware\Core\Checkout\Payment\Cart\PaymentHandler\PaymentHandlerType;
+use Shopware\Core\Checkout\Payment\Cart\PaymentTransactionStruct;
+use Shopware\Core\Checkout\Payment\PaymentException;
 use Shopware\Core\Framework\Context;
-use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\Framework\Struct\Struct;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
 
-class ExamplePayment implements RecurringPaymentHandlerInterface
+class MyCustomPaymentHandler extends AbstractPaymentHandler
 {
-    private OrderTransactionStateHandler $transactionStateHandler;
+    public function __construct(private readonly OrderTransactionStateHandler $transactionStateHandler)
+    {
+    }
 
-    public function __construct(OrderTransactionStateHandler $transactionStateHandler)
+    public function supports(PaymentHandlerType $type, string $paymentMethodId, Context $context): bool
     {
-        $this->transactionStateHandler = $transactionStateHandler;
+        // this payment handler supports recurring payments
+        return $type === PaymentHandlerType::RECURRING;
     }
 
     /**
-     * @throws RecurringPaymentProcessException
+     * This method is in the case of recurring payments only called during the initial checkout.
+     * You probably want to create a billing agreement between the customer and your PSP here.
+     * Do not forget to capture the initial charge as well.
      */
-    public function captureRecurring(RecurringPaymentTransactionStruct $transaction, Context $context): void
+    public function pay(Request $request, PaymentTransactionStruct $transaction, Context $context, ?Struct $validateStruct): ?RedirectResponse
+    {
+        // In here you should probably call your payment provider to create a billing agreement
+        // $this->myPaymentProvider->createBillingAgreement($transaction);
```

Comment:
maybe provide an insight, that the transaction can be recognized as recurring with the recurring struct in the `PaymentTransactionStruct`